python_modules/libraries/dagster-components/dagster_components_tests/code_locations/custom_sling_location/custom_sling_location/components/debug_sling_component/component.py

```diff
@@ -2,7 +2,7 @@
 
 from dagster._core.execution.context.asset_execution_context import AssetExecutionContext
 from dagster_components import component
-from dagster_components.lib.sling_replication import SlingReplicationComponent
+from dagster_components.lib.__extras__.sling.sling_replication import SlingReplicationComponent
```

Comment:
I think this is a pretty rough regression and not something we can work with long term if users are often going to customize these components -- is there a way we can hack around this to be along the lines of `from dagster_components.lib.sling import SlingReplicationComponent`?

Maybe w/ some hacking with a getattr in lib's __init__.py?

Reply:
And to be clear, I mean specifically for the import path here -- so keeping the __extras__ hierarchy in the source but hacking around it when we need to import the class

Reply:
Agree that owen that we can't ship this include path.

Reply:
If we want to avoid `__extras__` in the include path then I think it would be better to change the mechanism of module exclusion during traversal than hacking `getattr` (again, plays poorly with static analysis tools and thus UX). That could be done by having `lib/__init__.py` declare `__extras`:

```
### lib/__init__.py

__extras__ = [
    "dbt",
    "sling",
]
```

Then we could read the attribute `lib.__extras__` when starting the crawl and components wouldn't need to be nested under a `lib.__extras__` submodule.